backend/app/services/learning-service.py

```
@@ -0,0 +1,110 @@
+            'total_learned_mappings': len(learned_mappings),
+            'most_common_fields': [
+                {'field': mapping['field_name'], 'usage_count': mapping('usage_count')}
+                for mapping in learned_mappings[:10]
+                ],
+            'average_confidence': np.mean([m['confidence'] for m in learned_mappings]) if learned_mappings else 0.0,
+            'domain': domain
+                }
+
+    def export_learned_data(self, domain: str = None) -> Dict[str, Any]:
+        if domain:
+            mappings = self.config_service.get_learned_mappings(domains)
```

Ruff (0.12.2)

Undefined name `domains`

(F821)